transform.go

```diff
@@ -309,5 +309,47 @@ func transformImage(m image.Image, opt Options) image.Image {
 		m = imaging.FlipH(m)
 	}
 
+	// trim
+	if opt.Trim {
+		m = trimEdges(m)
+	}
+
 	return m
 }
+
+func trimEdges(img image.Image) image.Image {
+	bounds := img.Bounds()
+	minX, minY, maxX, maxY := bounds.Max.X, bounds.Max.Y, bounds.Min.X, bounds.Min.Y
+
+	// Get the color of the first pixel (top-left corner)
+	baseColor := img.At(bounds.Min.X, bounds.Min.Y)
+
+	// Check each pixel and find the bounding box of non-matching pixels
```

Comment:
If we assume that the border will only be a small-ish percentage of the edge of the image, then we should be able to optimize this a bit such that we don't have to inspect every pixel.  Fine to leave as-is for now, particularly if we really focus on making sure there is good test coverage... then future optimizations are a lot safer to make.